src/modules/m_nationalcodepage.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2014 Googolplexed <googol@googolplexed.net>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "inspircd.h"
+#include "caller.h"
+
+class ModuleNationalCodePage : public Module,public HandlerBase1<bool, const std::string&>
+{
+	caller1<bool, const std::string&> rememberer;
+	const unsigned char * lowermap_rememberer;
+	std::bitset<256> m_allowedchar;
+	unsigned char m_casemap[256];
+	std::string charset,casemapping;
+
+public:
+	ModuleNationalCodePage()
+		: rememberer(ServerInstance->IsNick), lowermap_rememberer(national_case_insensitive_map) {
+	}
+
+	//Check if Nick is valid using the current codepage
+	bool Call(const std::string& nick)
+	{
+		if(nick.length() > ServerInstance->Config->Limits.NickMax)
+			return false;
+
+		for(std::string::const_iterator it = nick.begin(); it != nick.end(); ++it)
+		{
+			//Check if character in normal range
+			if ((*it >= 'A') && (*it <= '}'))
```

Comment:
It's preferable to assign the result of `*it` to a variable once and inspect that instead of using the `*it` syntax in many places